### PR TITLE
Wtag SF 

### PR DESCRIPTION
**Pull request by siqiyyyy**

### CODE REVIEW

File: bucoffea/helpers/dataset.py [hunk 1]
```diff
@@ -24,6 +24,9 @@ def is_nlo_z(dataset):
 def is_nlo_w(dataset):
     return bool(re.match(r'W(\d*)Jet.*FXFX.*', dataset))
 
+def has_v_jet(dataset):
+    return bool(re.match(r'(WW|WZ|ZZ|WW).*', dataset))
```
Comment (AndreasAlbert) [inline bucoffea/helpers/dataset.py]: WW is there twice

File: bucoffea/monojet/monojetProcessor.py [hunk 2]
```diff
@@ -377,6 +379,16 @@ def process(self, df):
                 elif re.match(r'cr_g.*', region):
                     region_weights.add('trigger', np.ones(df.size))
 
+            if df['has_v_jet']:
+                if re.match(r'.*_loose_v.*', regions):
```
Comment (AndreasAlbert) [inline bucoffea/monojet/monojetProcessor.py]: this should be `region` rather than regions, same for the following lines

**Review by AndreasAlbert — CHANGES_REQUESTED** [review]

File: bucoffea/monojet/monojetProcessor.py (hunk 2 above)
Comment (siqiyyyy) [reply bucoffea/monojet/monojetProcessor.py]: That seems to be the bug I had, thanks!

File: bucoffea/helpers/dataset.py [hunk 3]
```diff
@@ -24,6 +24,9 @@ def is_nlo_z(dataset):
 def is_nlo_w(dataset):
     return bool(re.match(r'W(\d*)Jet.*FXFX.*', dataset))
 
+def has_v_jet(dataset):
+    return bool(re.match(r'(WW|WZ|ZZ).*', dataset))
```
Comment (AndreasAlbert) [inline bucoffea/helpers/dataset.py]: What about ttbar?

Comment (siqiyyyy) [reply bucoffea/helpers/dataset.py]: You are right, I should add that

File: bucoffea/plot/studies/wtag_sf/mistagSF.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+def divide_sumw2(sumw_a, sumw2_a, sumw_b, sumw2_b): #return (sumw_c, sumw2_c) for c=a/b
+    # check that the inputs have the same shape
+    assert len(set((sumw_a.shape, sumw_b.shape, sumw2_a.shape, sumw2_b.shape))) == 1
+    sumw_c  = np.zeros(sumw_a.shape)
+    sumw2_c = np.zeros(sumw2_a.shape)
+    for ientry in range(len(sumw2_c)):
+        if sumw2_b[ientry]>0:
+            sumw_c[ientry] = sumw_a[ientry] / sumw_b[ientry]
+            sumw2_c[ientry] = (sumw_a[ientry]**2 * sumw2_b[ientry] + sumw_b[ientry]**2 * sumw2_a[ientry]) / (sumw_b[ientry]**2 * sumw_b[ientry]**2)
```
Comment (AndreasAlbert) [inline bucoffea/plot/studies/wtag_sf/mistagSF.py:40]: @siqiyyyy just noticed this. When you are calculating an efficiency, the numerator and denominator share a large number of their events and are therefore not statistically independent. Can't use Gaussian error propagation in that case, but rather the clopper pearson interval, see e.g. https://github.com/CoffeaTeam/coffea/blob/master/coffea/hist/plot.py#L383

**Pull request merged**